skimage/measure/_ccomp.pyx

```diff
@@ -371,8 +371,10 @@ def label_cython(input_, neighbors=None, background=None, return_num=False,
         # use the full connectivity by default
         connectivity = ndim
     elif neighbors is not None:
-        DeprecationWarning("The argument 'neighbors' is deprecated, use "
-                           "'connectivity' instead")
+        # not sure why stacklevel should only be 2 not 3. Maybe cython
+        # is stripping away a stacklevel????
+        warn("The argument 'neighbors' is deprecated, use 'connectivity' "
+             "instead", stacklevel=2)
```

Comment:
add `category=DeprecationWarning`?

Reply:
No. Deprecation warnings get magically silenced by python. We don't do this consistently elsewhere in the library

Reply:
> get magically silenced by Python

?

```
 $ cat test.py
import warnings
warnings.warn('hello', category=DeprecationWarning)

 $ python test.py 
test.py:2: DeprecationWarning: hello
  warnings.warn('hello', category=DeprecationWarning)
```

> we don't do this consistently

never too late to start. ;)

Reply:
I think the previous code was simply incorrect, rather than any Python silencing.

Reply:
https://docs.python.org/3/library/warnings.html#warnings.warn

I can't send screenshot, but see the explanation there

Reply:
@hmaarrfk either I'm insane or that says nothing about DeprecationWarnings being ignored. It says that if `message` is already a DeprecationWarning object, whatever you give to category= will be ignored. But we are passing a string, so that's not an issue...?

Reply:
Note: by "insane" I mean "sleepy/tired". =P On a dark plane going cross country right now.

Reply:
Deprecation warning: Base category for warnings about deprecated features when those warnings are intended for other Python developers (ignored by default, unless triggered by code in __main__).

Reply:
@hmaarrfk :man_facepalming: Yikes. 

However, clicking on the link, I don't see the part in parentheses?

Reply:
I guess it is in their table only
![image](https://user-images.githubusercontent.com/90008/61674157-ad239780-acc0-11e9-8fa5-fd23d39a1c16.png)

was on my phone when i was sending the initial link.

Reply:
Wow. Just retested with #4035 and indeed the DeprecationWarning is silent. wth. Thanks for the pointer!!!

Reply:
Shouldn't we set a version at which the support for this deprecated argument is over? I don't see anything in TODO.txt.